src/base/static/components/input-form/index.js

```diff
@@ -271,9 +274,7 @@ class InputForm extends Component {
             {messages.validationHeader}
           </p>
           {Array.from(this.state.formValidationErrors).map((errorMsg, i) => (
-            <p key={i} className={"input-form__warning-msg"}>
-              {errorMsg}
-            </p>
+            <ErrorMessage key={i}>{errorMsg}</ErrorMessage>
```

Comment:
Note that I am refactoring this `<p>` element into a typography atom. In general, I think we should consider adopting the atomic design methodology. I see that we are doing that in our `ui-elements` dir, but atomic design provides a bit more definition and structure in its approach.

I'm definitely open to feedback about adopting atomic design, it really simplifies our testing and will make our code more reusable and testable, especially as we build out our component library. Feel free to read up more on atomic design here and lmk your thoughts:

http://bradfrost.com/blog/post/atomic-web-design/
https://about.futurelearn.com/blog/atomic-design-molecules-organisms
http://atomicdesign.bradfrost.com/chapter-1/

Reply:
I think this seems good. The only (potential) pitfall I can think of is if we need to do localization on a string that is composed of numerous typographic atoms, e.g.:

```
<SubmitterName /> <Action /> this <PlaceType />
```
which would render as something like `Joe created this place`.

I think i18next has a way of handling localization situations where the string to be localized is split across several components, so this might be fine. But if we have tons of typographic atoms I wonder how tenable that will be.

I'll investigate this a bit.

Reply:
UPDATE: This page on i18next has a discussion of how complex strings containing components are handled:

https://react.i18next.com/components/trans-component.html

I tested this approach with our submitter/action text string, and it seems to work well. The i18n strings that are produced are a little convoluted, but they can be generated automatically by i18next, so I don't think it's an issue.

Reply:
Cool - thanks for looking into that. Using atoms and small components in this manner is a common pattern, so we'll definitely need our translation pipeline to handle it.